src/gilded_rose.js

```diff
@@ -13,50 +13,82 @@ items.push(new Item('Sulfuras, Hand of Ragnaros', 0, 80));
-    if (items[i].name != 'Aged Brie' && items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-      if (items[i].quality > 0) {
-        if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-          items[i].quality = items[i].quality - 1
-        }
-      }
-    } else {
-      if (items[i].quality < 50) {
-        items[i].quality = items[i].quality + 1
-        if (items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-          if (items[i].sell_in < 11) {
-            if (items[i].quality < 50) {
-              items[i].quality = items[i].quality + 1
-            }
-          }
-          if (items[i].sell_in < 6) {
-            if (items[i].quality < 50) {
-              items[i].quality = items[i].quality + 1
-            }
-          }
-        }
-      }
-    }
-    if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-      items[i].sell_in = items[i].sell_in - 1;
-    }
-    if (items[i].sell_in < 0) {
-      if (items[i].name != 'Aged Brie') {
-        if (items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-          if (items[i].quality > 0) {
-            if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-              items[i].quality = items[i].quality - 1
-            }
-          }
-        } else {
-          items[i].quality = items[i].quality - items[i].quality
-        }
-      } else {
-        if (items[i].quality < 50) {
-          items[i].quality = items[i].quality + 1
-        }
-      }
-    }
+  items.forEach((item) => {updateItemQuality(item)})
+}
+
+function updateItemQuality(item) {
+  if(typeof itemInventory[item.name] !== 'function') {
+    regularDecrease(item)
+  } else {
+    itemInventory[item.name](item)
+  }
+}
+
+function increaseQuality(item, amt) {
+  amt = amt || 1
+  item.quality+=amt
+  if(item.quality > 50) {
+    item.quality = 50
+  }
+}
+
+function degradeQuality(item, amt) {
+  amt = amt || 1
+  item.quality-=amt
+  if(item.quality < 0) {
+    item.quality =0
   }
 }
+
+function increasesOverTime(item) {
+  decreaseDay(item)
+  item.sell_in < 0 ? increaseQuality(item, 2) : increaseQuality(item)
+}
+
+function increasesMoreTowardsDeadlineAndExpires(item) {
+  // decreaseDay(item)
+  // if(item.sell_in < 5) {}
+  //   increaseQuality(item, 3)
+  // } else if (item.sell_in < 10) {
+  //   increaseQuality(item, 2)
+  // } else {
+  //   increaseQuality(item)
+  // }
+  // if (item.sell_in < 0) {
+  //   item.quality = 0
+  // }
+  increaseQuality(item)
+  if (item.sell_in <= 10) {
+    increaseQuality(item)
+  }
+  if (item.sell_in <= 5) {
+    increaseQuality(item)
+  }
+  decreaseDay(item)
+  if (item.sell_in < 0) {
+    item.quality = 0
+  }
+}
+
+function regularDecrease(item) {
+  decreaseDay(item)
+  item.sell_in < 0 ? degradeQuality(item, 2) : degradeQuality(item)
+}
+
+function doubleDecreases(item) {
+  decreaseDay(item)
+  item.sell_in < 0 ? degradeQuality(item, 4) : degradeQuality(item, 2)
+}
+
+function decreaseDay(item) {
```

Comment:
Challenge: Can you write `one` function that can decrease sell_in **OR** quality?